src/entities/currency/lib.ts

```diff
@@ -24,7 +24,9 @@ export const createCurrencyAmountString = ({
   if (symbolPosition === "right") {
     parts.reverse();
   }
-  const currencyString = parts.join(hasSpaceBetweenAmountAndSymbol ? " " : "");
+  const currencyString = parts
+    .join(hasSpaceBetweenAmountAndSymbol ? " " : "")
+    .replace(/\B(?=(\d{3})+(?!\d))/g, ",");
```

Comment:
Move this regex to a constant

Reply:
To improve this, we can add a checkbox to the currency page for grouping numbers. Users can choose whether to group numbers or not. Additionally, please create unit tests for this regex, and make sure to test decimal numbers.